db2i_charsetSupport.cc

```diff
@@ -40,6 +40,65 @@ OF SUCH DAMAGE.
 #include "db2i_errors.h"
 #include "hash.h"
 
+struct EncodingMapEntry {
+  const char* csname;
+  const char* iconv_mariadb;
+  const char* iconv_db2;
+  unsigned short db2_ccsid;
+};
+
+// single byte is ebcdic single byte
+// multibyte is UTF-16
+// Here is the list of supported mariadb charsets
+// https://mariadb.com/kb/en/supported-character-sets-and-collations/
+static EncodingMapEntry encoding_map[] = {
+
+  {"utf8", "UTF-8", "UTF-8", 1208},
+  {"utf8mb4", "UTF-8", "UTF-8", 1208},
+  {"utf8mb3", "UTF-8", "UTF-8", 1208},
+  {"utf16", "UTF-16", "UTF-16", 1200},
+  {"ucs2", "UCS-2", "UCS-2", 13488},
+  {"ascii", "ISO8859-1", "IBM-1148", 1148}, // (US ASCII) use ISO8859-1?
+  {"latin1", "ISO8859-1", "IBM-1148", 1148}, // (cp1252 West European) AKA ISO8859-1
+  {"latin2", "ISO8859-2", "IBM-1153", 1153}, // (ISO 8859-2 Central European)
+  {"greek", "ISO8859-7", "IBM-4971", 4971}, // (ISO 8859-7 Greek) was 875
+  {"hebrew", "ISO8859-8", "IBM-424", 424}, // (ISO 8859-8 Hebrew)
+  {"latin5", "ISO8859-9", "IBM-1155", 1155}, // (ISO 8859-9 Turkish)
+  {"tis620", "TIS-620", "IBM-838", 838}, // (TIS620 Thai)
+
+  {"euckr", "EUC-KR", "UTF-16", 1200}, // (EUC-KR Korean)
+  {"gbk", "GBK", "UTF-16", 1200}, // (Simplified Chinese) found GBK_ct__64 & GBK_ct but unsure what ct means
+  {"gb2312", "GBK", "UTF-16", 1200}, // (Simplified Chinese) no converter found
+  {"sjis", "IBM-943", "UTF-16", 1200}, // (Shift-JIS Japanese) use IBM-943 ?
+  {"ujis", "EUC-JP", "UTF-16", 1200}, // (EUC-JP Japanese)
+  {"big5", "big5", "UTF-16", 1200}, // (Big5 Traditional Chinese)
+  {"cp1250", "IBM-1250", "IBM-1153", 1153}, // (Windows Central European)
+  {"cp1251", "IBM-1251", "IBM-1154", 1154}, // (Windows Cyrillic)
+  {"cp850", "IBM-850", "IBM-1148", 1148}, // (DOS West European)
+  {"cp852", "IBM-852", "IBM-1153", 1153}, // (DOS Central European)
+  {"cp1256", "IBM-1256", "IBM-420", 420}, // (Windows Arabic)
+  {"cp932", "IBM-943", "UTF-16", 1200}, // (SJIS for Windows Japanese)
+  {"macce", "IBM-1282", "IBM-1153", 1153}, // (Mac Central European)
+
+  // There are no iconv converter tables for the following charsets:
```

Comment:
Please add more clarity behind this comment. Does this mean users are just out of luck if using one of these? But maria supports them?

Reply:
Right as it is if any of the below charsets are used we will error out with: [DB2I_ERR_UNSUPP_CHARSET](https://github.com/zendtech/db2iengine/blob/master/db2i_errors.h#L58)

Reply:
Is this a behavior regression from previous versions of the engine? 
In either case, please update the comment with this information.

Reply:
> Is this a behavior regression from previous versions of the engine?

Good question, I found this table listing supported charsets on http://www.redbooks.ibm.com/redbooks/pdfs/sg247705.pdf

![image](https://user-images.githubusercontent.com/33973272/85483578-b41af000-b58a-11ea-9eb4-00c876facd90.png)


Looks like from our list of unsupported charsets the following were supported before

| charset   |    Db2 CCSID  |
| ----------   |  ------------  |
| cp866     |  880          |
| cp1257   | 1156         |
| latin7      | 1112          |

We should look into adding the above to the supported list.

Reply:
@ThePrez the old code would fail if it could not open a converter between the MariaDB CCSID and the Db2 CCSID. We looked through all the supported PASE iconv conversions and the ones below the comment are not available.